package/tests/test_api/test_graphql.py

```diff
@@ -522,8 +526,95 @@ def test_graphql_runs_tracking_data_endpoint(
         assert response.json() == expected_response
 
 
+class TestGraphQLMutation:
+    def test_update_user_details_success(
+        self, client, save_version, example_runs, example_db_dataset, mocker
+    ):
+        with mock.patch(
+            "kedro_viz.data_access.DataAccessManager.db_session",
+            new_callable=PropertyMock,
+        ) as mock_session:
+            mock_session.return_value = example_db_dataset
+            mocker.patch("kedro_viz.api.graphql.get_runs").return_value = example_runs
+            response = client.post(
+                "/graphql",
+                json={
+                    "query": """mutation {updateRunDetails(
+                        runId: "%s",
+                        runInput: {
+                        bookmark: false,
+                        title: "Hello Kedro", notes:
+                        "There are no notes"})
+                        {
+                            __typename
+                            ... on UpdateRunDetailsSuccess {
+                            runDetails
+                            }
+                            ... on BadInputType {
+                            runId
+                            errorMessage
+                            } }
+                            }"""
+                    % save_version
+                },
+            )
+            assert response.json() == {
+                "data": {
+                    "updateRunDetails": {
+                        "__typename": "UpdateRunDetailsSuccess",
+                        "runDetails": {
+                            "run_id": "2021-11-02T18.24.24.379Z",
+                            "bookmark": False,
+                            "title": "Hello Kedro",
+                            "notes": "There are no notes",
```

Comment:
nit: the `there are no notes` example is a bit counter intuitive in this positive, happy path.